ERCS/erc-7682.md

```diff
@@ -72,10 +72,118 @@ type AuxiliaryFundsCapability = {
 }
 ```
 
-### App Implementation
+### Extended Usage: `assetsRequiredForExecution` Parameter
+
+When a wallet indicates support for the `auxiliaryFunds` capability (i.e., `supported: true`), applications that use the [`wallet_sendCalls`](./eip-5792.md#wallet_sendcalls) method MAY include an `assetsRequiredForExecution` parameter in the `capabilities.auxiliaryFunds` object to enable the wallet to leverage this capability.
+
+This parameter is necessary because transaction call data alone does not reliably indicate which assets and amounts are needed for successful execution. Calls may involve complex logic or multiple contracts, making it difficult for wallets to infer requirements. By specifying required assets and amounts explicitly, apps ensure wallets have the information needed to provision, bridge, or swap assets as necessary for the transaction to succeed.
+
+### `wallet_sendCalls` Extended Parameter
+
+If included, the `assetsRequiredForExecution` parameter MUST be specified within the `capabilities.auxiliaryFunds` object of the `wallet_sendCalls` request.
+
+> **Note**: The `capabilities` object is specified in [EIP-5792](./eip-5792.md) where `wallet_sendCalls` is first defined: "The capabilities field is how an app can communicate with a wallet about capabilities a wallet supports. For example, this is where an app can specify a paymaster service URL from which an [ERC-4337](./eip-4337.md) wallet can request a paymasterAndData input for a user operation."
+
+```typescript
+// Example wallet_sendCalls type extension
+type WalletSendCallsRequest = {
+  version: string;
+  from: `0x${string}`;
+  chainId: `0x{string}`;
+  atomicRequired: boolean;
+  calls: Array<{
+    to: `0x${string}`;
+    data: string;
+    // ... other call fields ...
+  }>;
+  capabilities?: {
+    auxiliaryFunds?: {
+      assetsRequiredForExecution?: {
```

Comment:
Why can't this be named something simple like `required`? These capabilities already imply that they are intended "for execution", so the `*ForExecution` is kind of redundant.

Reply:
I'd be fine with that!

Reply:
Done here: https://github.com/ethereum/ERCs/pull/1126/commits/13bf16bad7223c59ae7c3dedf45b593c315cb0cd